lib/PrintAsObjC/PrintAsObjC.cpp

```diff
@@ -933,6 +933,9 @@ class ObjCPrinter : private DeclVisitor<ObjCPrinter>,
     } else {
       TypeDecl *typeDecl = declContext->getSelfNominalTypeDecl();
       
+      TypeMatchOptions matchMode = TypeMatchFlags::AllowOverride;
+      matchMode |= TypeMatchFlags::AllowTopLevelOptionalMismatch;
```

Comment:
I don't think you need AllowTopLevelOptionalMismatch for comparing just the types.